File: fixtures/polyglot/kotlin/orders/PaymentService.kt
package com.acme.kotlinorders

interface PaymentPort {
  fun charge(request: PaymentRequest): PaymentReceipt
}

class PaymentRequest(val orderId: String) {
}

class PaymentReceipt(val id: String) {
}

class PaymentService(private val invoices: InvoiceRepository) : PaymentPort {
  override fun charge(request: PaymentRequest): PaymentReceipt {
    return invoices.create(request)
  }
}

class InvoiceRepository {
  fun create(request: PaymentRequest): PaymentReceipt = PaymentReceipt(request.orderId)
}

File: fixtures/polyglot/php/src/Orders/PaymentService.php
<?php

namespace Acme\Orders;

use Acme\Billing\InvoiceRepository;

interface PaymentPort
{
    public function charge(PaymentRequest $request): PaymentReceipt;
}

final class PaymentService implements PaymentPort
{
    private InvoiceRepository $invoices;

    public function __construct(InvoiceRepository $invoices)
    {
        $this->invoices = $invoices;
    }

    public function charge(PaymentRequest $request): PaymentReceipt
    {
        return $this->invoices->create($request);
    }
}

final class PaymentReceipt
{
    public function __construct(public string $id) {}
}

final class PaymentRequest
{
    public function __construct(public string $orderId) {}
}
